client/state/jetpack/onboarding/test/reducer.js

```diff
@@ -0,0 +1,140 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import deepFreeze from 'deep-freeze';
+
+/**
+ * Internal dependencies
+ */
+import { credentials as credentialsReducer, reducer } from '../reducer';
+import {
+	DESERIALIZE,
+	JETPACK_CONNECT_AUTHORIZE_RECEIVE,
+	JETPACK_ONBOARDING_CREDENTIALS_RECEIVE,
+	SERIALIZE,
+} from 'state/action-types';
+import { useSandbox } from 'test/helpers/use-sinon';
+
+describe( 'reducer', () => {
+	useSandbox( sandbox => {
+		sandbox.stub( console, 'warn' );
```

Comment:
> Let's use Jest, and can we mock on a per-test basis? When we expect and want to hide warnings, let's do that, otherwise warnings can be helpful to see during tests.

Awesome, I tried mocking on a per-test basis using the old sandbox mock while working on this PR but couldn't make it work. Works like a charm with Jest! 26ce34e